lib/utils.js

```diff
@@ -304,4 +310,32 @@ utils.mapHeaders = function mapHeaders(headers, addHeaders) {
     return headers;
 };
 
+utils.isUnsupportedQuery = function isUnsupportedQuery(queryStr) {
+    if (!queryStr) {
+        return false;
+    }
+    // Reject unsupported query fields (e.g. ?accelerate), but allow
+    // the same strings as values (e.g. ?test=accelerate) and queries like
+    // '?acceleratefoobar' or '&foobaraccelerate'
+    return constants.unsupportedQueries.some(q => {
```

Comment:
I extracted the functions and ran some timing tests locally to try and compare the performance between checking the querystr and the query object. It seems checking values for the query object is more performant.

with 1000 time measurements per median, using a querystr of 400 queries (e.g. `testquery0&testquery1&[...]`:
1. fxn using querystr took (medians) [ 127385, 127844, 125469, 125785, 125371 ] nanoseconds, with an average of 126370.8
2. fxn using queryobj took [ 64499, 75429, 68070, 77557, 74643 ] nanoseconds, with an average of 72039.6

with 100,000 time measurements per median:
1. fxn using querystr took [ 126757, 125313, 126057, 126333, 124928 ] nanoseconds, with an average of 125877.6
2. fxn using queryobj took [ 74821, 68282, 72476, 68157, 74749 ] nanoseconds, with an average of 71697

I also tried to compare the two codebases using the apache benchmarking tool against the server running the respective code, but testing with 10,000 requests (100 concurrent requests) and querystring of length 20, 100, and 400 didn't seem to reveal conclusive differences. Since more testing is required to get something conclusive with `ab`, we'll go with the results from extracting the functions. Query object wins :1st_place_medal: